### DISCUSSION TITLE
Prisma queries

### DISCUSSION THREAD

**Discussion by elma-webdev** [prisma/prisma#25421] [Q&A] [answered]
### Question

I want to develop API with prisma, typeScript and nodeJs. I have three models in Prisma  named user, task and userTask (relationship many to many explicit). 
To create a userTask they must verify if the user and task exists and if they exist, I want to see how many tasks a user have. If a user just have three tasks he cannot make more tasks per day, unless he concludes one of his tasks.
So which command is the most correct to use? I tried to use include, select and where... I make confused, I don't know how to use the command count and something like that.
I just have the controllers done, but is missing this feature
HELP PLZ

![Captura de Tela (1180)](https://github.com/user-attachments/assets/3e217da1-44c9-457f-b16b-262835d5eac0)
![Captura de Tela (1181)](https://github.com/user-attachments/assets/d930d052-d6ac-4115-832a-80a20652b8a0)

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @elma-webdev 👋 

To implement the feature you described, you need a query that checks whether the `user` and `task` exist, counts the number of active tasks a user has, and enforces the limit of three tasks unless one is completed. The steps to implement this will look like:

1. Check if the user exists.
2. Check if the task exists.
3. Count the number of active tasks (not completed) for the user.
4. Enforce the task limit (max 3 active tasks per day unless one is completed).

The code will look something like:
```ts
import { PrismaClient } from '@prisma/client';
const prisma = new PrismaClient();

export const createUserTask = async (req, res) => {
  const { userId, taskId } = req.body;

  // Step 1: Verify if the user exists
  const user = await prisma.user.findUnique({
    where: { id: userId }
  });

  if (!user) {
    return res.status(404).json({ error: "User not found" });
  }

  // Step 2: Verify if the task exists
  const task = await prisma.tasks.findUnique({
    where: { id: taskId }
  });

  if (!task) {
    return res.status(404).json({ error: "Task not found" });
  }

  // Step 3: Count active (PENDENT) tasks for the user
  const activeTasksCount = await prisma.userTasks.count({
    where: {
      idUser: userId,
      status: 'PENDENT' // Assuming 'PENDENT' means the task is not completed
    }
  });

  // Step 4: Enforce the task limit (max 3 active tasks)
  if (activeTasksCount >= 3) {
    return res.status(403).json({
      error: "Task limit reached. Complete a task to add a new one."
    });
  }

  // Step 5: If the user can take on another task, create the UserTask entry
  const newUserTask = await prisma.userTasks.create({
    data: {
      idUser: userId,
      idTasks: taskId,
      points: 10,  // You can set default values here if needed
      status: 'PENDENT',
      finishedAt: null // You can set this once the task is completed
    }
  });

  return res.status(201).json(newUserTask);
};
```

**Reply by elma-webdev** [reply]
From the bottom of my heart, thank yoi very much. But actually I found a way to solve this issue, not so simples as you had done, but if works! Thanks again!

**Reply by ludralph** [reply]
I'm glad to have helped.
